docs/reference/ml/ml-shared.asciidoc

```diff
@@ -669,13 +687,20 @@ decision tree when the tree is trained.
 end::dfas-num-splits[]
 
 tag::dfas-soft-limit[]
-Tree depth limit is used for calculating the tree depth penalty. This is a soft
-limit, it can be exceeded.
+Advanced configuration option. {ml} uses loss guided tree growing.
+This means that trees will grow where the regularized loss reduces
+the most. The regularized loss increases quickly where the tree depth
+exceeds this parameter. This is a soft limit, it can be exceeded.
+Values must be greater than or equal to 0.
+By default, this value is calculated during hyperparameter optimization.
 end::dfas-soft-limit[]
 
 tag::dfas-soft-tolerance[]
-Tree depth tolerance is used for calculating the tree depth penalty. This is a
-soft limit, it can be exceeded.
+Advanced configuration option.
+This controls how quickly the regularized loss increases when the tree
+depth exceeds {dfa-soft-limit}.
```

Comment:
{dfa-soft-limit} won't resolve as it is not a defined attribute in [shared/attributes.asciidoc](https://github.com/elastic/docs/blob/master/shared/attributes.asciidoc).

```suggestion
depth exceeds `soft_tree_depth_limit`.
```